controller/challenge-controller.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const debug = require('debug')('ramble:challenge-controller');
+const Challenge = require('../model/challenge');
+const httpErrors = require('http-errors');
+
+exports.createChallenge = function(challengeData) {
+  debug('createChallenge');
+  return new Promise((resolve, reject) => {
+    new Challenge(challengeData).save()
+    .then((challenge) => {
+      resolve(challenge);
+    })
+    .catch(err => reject(httpErrors(400, err.message)));
+  });
+};
+
+exports.fetchChallenge = function(id) {
+  debug('fetchChallenge');
+  return new Promise((resolve, reject) => {
+    Challenge.findOne({_id: id})
+    .then(challenge => resolve(challenge))
+    .catch(err => reject(httpErrors(404, err.message)));
+  });
+};
+
+exports.fetchAllChallenges = function(){
+  debug('fetchAllChallenges');
+  return new Promise((resolve, reject) => {
+    Challenge.find({})
+    .then(challenges => resolve(challenges))
+    .catch(err => reject(httpErrors(404, err.message)));
+  });
+};
+
+exports.updateChallenge = function(id, data) {
+  debug('updateChallenge');
+  return new Promise((resolve, reject) => {
+    if(!id){
+      var err = httpErrors(400, err.message);
+      return reject(err);
+    }
+    if(!data){
+      err = httpErrors(400, err.message);
+      return reject(err);
+    }
+    Challenge.findOne({_id: id})
```

Comment:
In stead of using Challenge.findOne you could be using the ^ function fetchChallenge above, and then that would do alot of error handling for you!